src/components/Bag/classes.js

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import { v4 as uuidv4 } from "uuid";
+import { Label, InsertDriveFile, Folder } from "@material-ui/icons";
+
+class BagItem {
+    constructor(item) {
+        this.item = item;
+    }
+
+    icon(t) {
+        return <Label />;
+    }
+
+    get label() {
+        return (
+            this.item?.label ||
+            this.item?.name ||
+            this.item?.description ||
+            "override label() please"
+        );
+    }
+
+    get shareable() {
+        return false;
+    }
+
+    get downloadable() {
+        return false;
+    }
+
+    get id() {
+        return this.item.id;
+    }
+}
+
+class FileBagItem extends BagItem {
+    icon(t) {
+        return <InsertDriveFile />;
+    }
+
+    get shareable() {
+        return true;
+    }
+
+    get downloadable() {
+        return true;
+    }
+
+    get name() {
+        return this.item.path.substring(this.item.path.lastIndexOf("/") + 1);
```

Comment:
In `/src/components/data/utils`, we have a function `parseNameFromPath` that does something similar.  It'd be good if we used the same function here, as well as for `FolderBagItem`